operator/autoscaler.go

```diff
@@ -50,86 +50,108 @@ func noopScalingOperation(description string) *ScalingOperation {
 	}
 }
 
-func getScalingDirection(eds *zv1.ElasticsearchDataSet, esMSet *zv1.ElasticsearchMetricSet, metricsInterval time.Duration) ScalingDirection {
-	scaling := eds.Spec.Scaling
-	name := eds.Name
-	namespace := eds.Namespace
+type AutoScaler struct {
+	logger          *log.Entry
+	eds             *zv1.ElasticsearchDataSet
+	esMSet          *zv1.ElasticsearchMetricSet
+	metricsInterval time.Duration
+	pods            []v1.Pod
+	esClient        *ESClient
+}
+
+func NewAutoScaler(es *ESResource, metricsInterval time.Duration, esClient *ESClient) *AutoScaler {
```

Comment:
Question: Does is make sense to inject the KubeRecorder here? 
https://github.com/zalando-incubator/es-operator/blob/ffe837c98377c3e7d6abe61b7cd4c5ba5bdfd60e/operator/elasticsearch.go#L86

This would allow us to write autoscaling events on the EDS. I'm not opinionated here, just an idea.

Reply:
yes, sounds good. I'd keep it out of the scope of this PR, though. We can create a follow-up item.